components/queryString/QueryStringForm.tsx

```diff
@@ -0,0 +1,154 @@
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import useSupportsClipboardRead from '../../hooks/useSupportsClipboardRead';
+import { ToastProps } from '../Toast';
+import QueryParamsForm, {
+  defaultParams,
+  QueryParams,
+} from './QueryParamsForm';
+
+export interface QueryStringFormProps {
+  queryString: string | void;
+  queryParams: QueryParams;
+  error: string;
+  setQueryString: (_queryString: string | void) => void;
+  setQueryParams: (_queryParams: QueryParams) => void;
+  setError: (_error: string) => void;
+  setToastMessage: (_toastMessage: string) => void;
+  setToastOpen: (_toastOpen: boolean) => void;
+  setToastSeverity: (_toastSeverity: ToastProps['severity']) => void;
+}
+
+export default function QueryStringForm({
+  queryString,
+  setQueryString,
+  queryParams,
+  setQueryParams,
+  error,
+  setError,
+  setToastMessage,
+  setToastOpen,
+  setToastSeverity,
+}: QueryStringFormProps) {
+  const supportsClipboardRead = useSupportsClipboardRead();
+  const { t } = useTranslation(['queryString', 'common']);
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const { value } = event.target;
+    setQueryString(value);
+    const newParams: QueryParams = [];
+    const params = new URLSearchParams(value.split('?')?.[1] || value);
+    params.forEach((val, key) => {
+      newParams.push({
+        param: key,
+        value: val,
+        key,
+      });
+    });
+    newParams.push({ param: '', value: '', key: 'last' });
+    setQueryParams(newParams);
+  }
+
+  return (
+    <Box
+      display='flex'
+      flexDirection='column'
+      justifyContent='stretch'
+      gap={6}
+      width={1000}
+      maxWidth='100%'
+    >
+      <Box
+        display='flex'
+        flexDirection='column'
+        gap={2}
+      >
+        <TextField
+          multiline
+          error={error.length > 0}
+          label='Query String'
+          name='queryString'
+          onChange={handleChange}
+          value={queryString}
+        />
+        <Box
+          display='flex'
+          flexWrap='wrap'
+          justifyContent='end'
+          gap={2}
+        >
+          <Button
+            startIcon={<ClearIcon />}
+            disabled={!queryString}
+            onClick={() => {
+              setQueryString('');
+              setQueryParams(defaultParams);
+            }}
+          >
+            {t('common:clear')}
+          </Button>
+          <Button
+            startIcon={<ContentCopyIcon />}
+            disabled={!queryString}
+            name='Copy'
+            onClick={() => {
+              navigator.clipboard.writeText(queryString || '').then(
+                () => {
+                  setToastMessage('Copied to clipboard');
```

Comment:
Please the common translation fields for the toast messages, there is one for "copied to clipboard" and "failed to copy"